IconView.swift

```diff
@@ -0,0 +1,200 @@
+        trianglePath.addLine(to: CGPoint(x: mid + 2.5, y: circleViewHeight))
+        trianglePath.addLine(to: CGPoint(x: mid, y: circleViewHeight + triangleHeight))
+        trianglePath.close()
+        
+        let shapeLayer = CAShapeLayer()
+        shapeLayer.path = trianglePath.cgPath
+        
+        shapeLayer.fillColor = UIColor(red: 47/255, green: 49/255, blue: 53/255, alpha: 1.0).cgColor
+        shapeLayer.strokeColor = UIColor(red: 47/255, green: 49/255, blue: 53/255, alpha: 1.0).cgColor
+        shapeLayer.lineWidth = desiredLineWidth
+        
+        return shapeLayer
+    }
+    
+    internal func drawGroundBlackCircle(yPos:CGFloat) -> CAShapeLayer {
+        let desiredLineWidth:CGFloat = 1    // your desired value
+        
+        let circlePath = UIBezierPath(
+            arcCenter: CGPoint(x: self.frame.size.width/2.0, y:yPos),
+            radius: CGFloat(5),
+            startAngle: CGFloat(0),
+            endAngle:CGFloat(M_PI * 2),
+            clockwise: true)
+        
+        let shapeLayer = CAShapeLayer()
+        shapeLayer.path = circlePath.cgPath
+        
+        shapeLayer.fillColor = UIColor.black.cgColor
+        shapeLayer.strokeColor = UIColor.black.cgColor
+        shapeLayer.lineWidth = desiredLineWidth
+        
+        return shapeLayer
+    }
+    
+    internal func drawGroundWhiteCircle(yPos:CGFloat) -> CAShapeLayer {
+        let desiredLineWidth:CGFloat = 1    // your desired value
+        
+        let circlePath = UIBezierPath(
+            arcCenter: CGPoint(x: self.frame.size.width/2.0, y:yPos),
+            radius: CGFloat(8),
+            startAngle: CGFloat(0),
+            endAngle:CGFloat(M_PI * 2),
+            clockwise: true)
+        
+        let shapeLayer = CAShapeLayer()
+        shapeLayer.path = circlePath.cgPath
+        
+        shapeLayer.fillColor = UIColor(red: 239/255, green: 239/255, blue: 239/255, alpha: 1.0).cgColor
+        shapeLayer.strokeColor = UIColor(red: 239/255, green: 239/255, blue: 239/255, alpha: 1.0).cgColor
+        shapeLayer.lineWidth = desiredLineWidth
+        
+        return shapeLayer
+    }
+    
+    internal func drawGroundRedCircle(yPos:CGFloat) -> CAShapeLayer {
+        let desiredLineWidth:CGFloat = 1    // your desired value
+        
+        let circlePath = UIBezierPath(
+            arcCenter: CGPoint(x: self.frame.size.width/2.0, y:yPos),
+            radius: CGFloat(13),
+            startAngle: CGFloat(0),
+            endAngle:CGFloat(M_PI * 2),
+            clockwise: true)
+        
+        let shapeLayer = CAShapeLayer()
+        shapeLayer.path = circlePath.cgPath
+        
+        shapeLayer.fillColor = UIColor(red: 206/255, green: 73/255, blue: 55/255, alpha: 1.0).cgColor
+        shapeLayer.strokeColor = UIColor(red: 206/255, green: 73/255, blue: 55/255, alpha: 1.0).cgColor
+        shapeLayer.lineWidth = desiredLineWidth
+        
+        return shapeLayer
+    }
```

Comment:
It seems like all the draw circle functions have similar repetitive code, so we should refactor them to use a shared function